libraries/SRV_Channel/SRV_Channel_aux.cpp

```diff
@@ -477,6 +477,16 @@ bool SRV_Channels::find_channel(SRV_Channel::Aux_servo_function_t function, uint
     return false;
 }
 
+// find first channel that a function is assigned to, return -1 on failure
```

Comment:
Return false, and do it by reference, would let scripting do nil punning and stay consistent. Nil punning will only work if we routinely embrace it.

Reply:
I did try to make that work but couldn't. The runtime kept telling me I had the wrong number of arguments. I suspect another bug in the generator

Reply:
Nil punning is actually working on other functions at the moment (and in the CI suite now), so I'm much less confident in that conclusion :)

Reply:
can you give me an example on how to use it for find_channel() that works?

Reply:
``` diff
diff --git a/libraries/AP_Scripting/generator/description/bindings.desc b/libraries/AP_Scripting/generator/description/bindings.desc
index 94d71baacc..29d03d13fd 100644
--- a/libraries/AP_Scripting/generator/description/bindings.desc
+++ b/libraries/AP_Scripting/generator/description/bindings.desc
@@ -152,5 +152,5 @@ singleton AP_SerialLED method send void
 
 include SRV_Channel/SRV_Channel.h
 singleton SRV_Channels alias SRV_Channels
-singleton SRV_Channels method find_channel int8_t SRV_Channel::ServoFunction'enum SRV_Channel::k_none SRV_Channel::k_nr_aux_servo_functions-1
+singleton SRV_Channels method find_channel boolean SRV_Channel::Aux_servo_function_t'enum SRV_Channel::k_none SRV_Channel::k_nr_aux_servo_functions-1 uint8_t'Null
 

```